src/decorators.ts

```diff
@@ -1,15 +1,31 @@
-import {Container} from "./Container";
+
+import {Container, ServiceDescriptor} from "./Container";
+
 
 /**
  * Marks class as a service that can be injected using container.
  *
- * @param name Optional service name can be specified. If service name is specified then this service can only be
- *              retrieved by a service name. If no service name is specified then service can be retrieved by its type
+ * @param descriptor
  */
-export function Service(name?: string) {
+export function Service(descriptor: ServiceDescriptor = {}) {
```

Comment:
This will make the code more complex and harder to read. I'm not really sure how you want it to be.